tests/test_initialize_simulation.py

```diff
@@ -54,6 +54,19 @@ def filter_station(
         self.assertIsNone(sim.bases.get("b1"), "should not have loaded base b1")
         self.assertIsNone(sim.stations.get("s1"), "should not have loaded station s1")
 
+    def test_load_simulation_with_custom_instruction_function(self):
+        conf = mock_config().suppress_logging()
+
+        def custom_instruction_function(
+            sim: SimulationState, env: Environment
+        ) -> Tuple[Instruction, ...]:
+            dummy_instructions = (IdleInstruction("v1"), IdleInstruction("v2"))
+            return dummy_instructions
+
+        rp = load_simulation(conf, custom_instruction_generators=[custom_instruction_function])
+
+        result = crank(rp, 1)
+
```

Comment:
this test should include an assertion, like, perhaps the custom function creates a fixed set of instructions that can be confirmed at the end

Reply:
yeah that's a good call, I updated the test to make sure the instructions are getting applied and using two different functions to make sure the name property is working